builder/vsphere/common/step_hardware.go

```diff
@@ -48,6 +48,16 @@ type HardwareConfig struct {
 func (c *HardwareConfig) Prepare() []error {
 	var errs []error
 
+	if c.CPUCoresPerSocket != 0 && c.CPUs != 0 {
+		if c.CPUs%c.CPUCoresPerSocket != 0 {
+			errs = append(errs, fmt.Errorf("'CPUs' must be devidable by 'cpu_cores'"))
+		}
```

Comment:
I'm confused with this verification. I was expecting that each CPU socket, the number defined with `CPUs`, would get the number of cores defined in `cpu_cores `. Why CPUs must be dividable by cpu_cores?

Reply:
Cause in result `CPUs = CoresPerSocket * Sockets`, in ESXi and Packer it's possible to change `CPUs` and `CoresPerSocket`. 

PS: Didn't notice spelling issue :)

Reply:
Looking at the docs, I think what we call CPUs is actually the sockets.